shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -350,7 +344,12 @@ private void setSystemChromeEnabledSystemUIOverlays(
    * PlatformPlugin}.
    */
   public void updateSystemUiOverlays() {
-    activity.getWindow().getDecorView().setSystemUiVisibility(mEnabledOverlays);
+    setSystemChromeEnabledSystemUIMode(currentSystemUiMode);
+
+    if (currentOverlays != null) {
+      setSystemChromeEnabledSystemUIOverlays(currentOverlays);
```

Comment:
`setSystemChromeEnabledSystemUIOverlays` sets `currentOverlays`. 

Is what situation `updateSystemUiOverlays` needs to call setSystemChromeEnabledSystemUIOverlays again with the same value?

Reply:
If someone decides to make a call to `setSystemChromeEnabledSystemUIMode` after `updateSystemUiOverlays` is called to change the `systemUiMode`, this could possibly hide some of the system bars, so then calling `updateSystemUiOverlays` would show those system bars again.

Reply:
right, but wouldn't an external call to `setSystemChromeEnabledSystemUIMode` set a new `currentSystemUiMode ` , thus making `updateSystemUiOverlays()` call `setSystemChromeEnabledSystemUIMode(currentSystemUiMode)` with the same value?

Reply:
Oh, I see what you mean. In that case, if `currentOverlays` isn't null, then there's no need to make the call to `setSystemChromeEnabledSystemUIMode`.